integration-cli/docker_api_swarm_service_test.go

```diff
@@ -175,6 +175,115 @@ func (s *DockerSwarmSuite) TestAPISwarmServicesUpdate(c *check.C) {
 		map[string]int{image1: instances})
 }
 
+func (s *DockerSwarmSuite) TestAPISwarmServicesUpdateStartFirst(c *check.C) {
+	d := s.AddDaemon(c, true, true)
+
+	// service image at start
+	image1 := "busybox:latest"
+	// target image in update
+	image2 := "testhealth"
+
+	// service started from this image won't pass health check
+	_, _, err := d.BuildImageWithOut(image2,
+		`FROM busybox
+		HEALTHCHECK --interval=1s --timeout=1s --retries=1024\
+		  CMD cat /status`,
+		true)
+	c.Check(err, check.IsNil)
+
+	// create service
+	instances := 5
+	parallelism := 2
+	rollbackParallelism := 3
+	id := d.CreateService(c, serviceForUpdate, setInstances(instances), setUpdateOrder(swarm.UpdateOrderStartFirst), setRollbackOrder(swarm.UpdateOrderStartFirst))
+
+	checkStartingTasks := func(expected int) []swarm.Task {
+		var startingTasks []swarm.Task
+		waitAndAssert(c, defaultReconciliationTimeout, func(c *check.C) (interface{}, check.CommentInterface) {
+			tasks := d.GetServiceTasks(c, id)
+			startingTasks = nil
+			for _, t := range tasks {
+				if t.Status.State == swarm.TaskStateStarting {
+					startingTasks = append(startingTasks, t)
+				}
+			}
+			return startingTasks, nil
+		}, checker.HasLen, expected)
+
+		return startingTasks
+	}
+
+	makeTasksHealthy := func(tasks []swarm.Task) {
+		for _, t := range tasks {
+			containerID := t.Status.ContainerStatus.ContainerID
+			d.Cmd("exec", containerID, "touch", "/status")
+		}
+	}
+
+	// wait for tasks ready
+	waitAndAssert(c, defaultReconciliationTimeout, d.CheckRunningTaskImages, checker.DeepEquals,
+		map[string]int{image1: instances})
+
+	// issue service update
```

Comment:
This is admittedly quite nit picky, but I personally don't find these inline comments provide much value. Many of them just echo the function name ("make it healthy", "wait for tasks ready", "update service"). These can lead to confusion when the code is modified and the comments no longer reflect reality.

I subscribe to the school of thought that inline comments should generally be avoided. They can be used sparingly when there is a very unexpected line that needs explanation. Instead of inline comments, code should be structured to reveal it's intent. Comments on functions and structs on the other hand are great, because they can be used to generate documentation, and they apply to a well defined block of code (the function body or struct definition).

Applying this rule here, some of these comments could be deleted (as the function names are already explicit enough), others (like "wait for tasks ready") could be replaced by extracting a new function, which would also remove duplication:

```go
func waitForTasksReady(c *check.C, tasks map[string]int) {
   ...
}
```

Are there benefits to this style of commenting that I'm missing?